### PR TITLE
[CARBONDATA 1040] Add notes of unsupported update table and delete records in spark 2.1

### PR DESCRIPTION
**Pull request by chenerlu**
Description: Add description of unsupported update table and delete records in spark 2.1

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1976/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1979/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1980/

**Comment by sgururajshetty** [comment]
LGTM
@chenliang613 kindly review

**Comment by jbonofre** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2069/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2072/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2073/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2074/

File: docs/dml-operation-on-carbondata.md [hunk 1]
```diff
@@ -340,6 +340,8 @@ SET (column_name1, column_name2,) =
 | sourceColumn | The source table column values to be updated in destination table. |
 | sourceTable | The table from which the records are updated into destination Carbon table. |
 
+NOTE: This functionality is currently not supported in Spark 2.x and may support in future.
```
Comment (chenerlu) [inline docs/dml-operation-on-carbondata.md]: Done, please help to merge this PR, thx.

**Pull request closed without merging**